src/ArborX_DistributedSearchTree.hpp

```diff
@@ -118,12 +119,21 @@ DistributedSearchTree<MemorySpace, Enable>::DistributedSearchTree(
 
   // Create new context for the library to isolate library's communication from
   // user's
-  MPI_Comm_dup(comm, &_comm);
+  _comm_ptr.reset(
+      [comm]() {
+        auto p = std::make_unique<MPI_Comm>();
+        MPI_Comm_dup(comm, p.get());
+        return p.release();
+      }(),
+      [](MPI_Comm *p) {
+        MPI_Comm_free(p);
+        delete p;
+      });
```

Comment:
Please add comments explaining the reasons behind this complication, including what problems it is trying to solve.